File: src/KeyGenerator.java
import java.util.Random;
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

public class KeyGenerator {
    private static final String characterPool = "aAbBcCdDeEfFgGhHiIjJkKlLmMnNoOpPqQrRsStTuUvVwWxXyYzZ";
    

    public static List<Integer> generateUniqueKeys(int rounds) {
        List<Integer> keys = new ArrayList<>();
        
        Collections.addAll(keys, 0b0110001111011110, 0b0110001111011111, 0b0110001111011011);
        return keys;
    }

    public static int generateRailFenceCipherKey() {
        return 0b0000000011011011;
        // return value << 8;
    }


}

File: py_src/key_generator.py
import random

class KeyGenerator:
    SEED = 98942897316645566
    CHARACTER_POOL = "aAbBcCdDeEfFgGhHiIjJkKlLmMnNoOpPqQrRsStTuUvVwWxXyYzZ"

    @staticmethod
    def generate_unique_keys(rounds):
        random.seed(KeyGenerator.SEED)
        keys = []

        for i in range(rounds):
            first_char = KeyGenerator.CHARACTER_POOL[random.randint(0, len(KeyGenerator.CHARACTER_POOL) - 1)]
            second_char = KeyGenerator.CHARACTER_POOL[random.randint(0, len(KeyGenerator.CHARACTER_POOL) - 1)]
            input = (ord(first_char) << 8) | ord(second_char)

            if i == 0:
                response = KeyGenerator.generate_rail_fence_cipher_key(input)
            elif i == 1:
                response = KeyGenerator.generate_columnar_cipher_key(input)
            else:
                response = KeyGenerator.generate_vigenere_cipher_key(input)

            keys.append(response)
        return keys

    @staticmethod
    def generate_rail_fence_cipher_key(input):
        return input %len(KeyGenerator.CHARACTER_POOL)

    @staticmethod
    def generate_columnar_cipher_key(input):
        return int('{:032b}'.format(input)[::-1], 2)

    @staticmethod
    def generate_vigenere_cipher_key(input):
        return ~input
